viewer/core/src/datamodels/cad/rendering/EffectRenderManager.ts

```diff
@@ -508,6 +509,16 @@ export class EffectRenderManager {
       result.hasBackElements = result.hasBackElements || hasBackElements;
       result.hasInFrontElements = result.hasInFrontElements || hasInFrontElements;
       result.hasGhostElements = result.hasGhostElements || hasGhostElements;
+    });
+
+    // Split scenes based on what render stages we need
+    const { hasBackElements, hasInFrontElements, hasGhostElements } = result;
+    this._rootSectorNodeBuffer.forEach(rootSectorNodeData => {
```

Comment:
So the bug fix was to compute `hasBackElements` etc. for every `rootSectorNodeData` and then using the accumulated result for determining what to show?

Reply:
Yes